python/tank/log.py

```diff
@@ -235,6 +237,125 @@ class LogManager(object):
     # keeps track of the single instance of the class
     __instance = None
 
+    class _SafeRotatingFileHandler(logging.handlers.RotatingFileHandler):
+        """
+        Provides all the functionality provided by Python's built-in RotatingFileHandler, but with a
+        failsafe when an I/O error happens when doing the rollover. In that case, the failure to
+        rename files will be ignored and the handler will keep writing to the current file. A message
+        will also be logged at the exception level so the user is aware that something really bad just
+        happened. Finally, the handler will not try to rollover in the future.
```

Comment:
Can we add something like

The consequence of this may be that the log files exceed the specified maximum size.